### ISSUE TITLE
Question: about length of loader

### ISSUE THREAD

**Issue by ShowLo** [NVIDIA/DALI#1427]
In https://github.com/NVIDIA/DALI/blob/master/docs/examples/pytorch/resnet50/main.py, we could find that the length of a loader is, eg, `train_loader_len = int(train_loader._size / args.batch_size)`. However, I think that it should be math.ceil(train_loader._size / args.batch_size), otherwise the last iteration is ignored, and that in the following `adjust_learning_rate(optimizer, epoch, i, train_loader_len)`, the lr will be wrong.

**Comment by JanuszL** [comment]
Hi,
You are correct, the value is not precise (fixed in https://github.com/NVIDIA/DALI/pull/1428). Please notice that it is used only to adjust a learning rate. The real number of iterations is obtained by `enumerate(train_loader)`.
So in the end warmup will differ a bit but the whole trainings still lasts a proper number of iterations.

**Comment by ShowLo** [comment]
> Hi,
> You are correct, the value is not precise (fixed in #1428). Please notice that it is used only to adjust a learning rate. The real number of iterations is obtained by `enumerate(train_loader)`.
> So in the end warmup will differ a bit but the whole trainings still lasts a proper number of iterations.

Thanks for your reply, and I have another question. The same in `main.py`, we could see `train_loader = DALIClassificationIterator(pipe, size=int(pipe.epoch_size("Reader") / args.world_size))`, since I couldn't find the definition of `DALIClassificationIterator`, the meaning of the parameter `size` is unclear to me. I know it work well when `world_size=1`, and I'm just curious about whether it will also work right when `world_size>1`, e.g., `pipe.epoch_size('Reader')=3` and `world_size=2`, that `size=1`, is this correct?

**Comment by JanuszL** [comment]
You can read more about `DALIClassificationIterator` [here](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/plugins/pytorch_plugin_api.html#module-nvidia.dali.plugin.pytorch).
To be supper accurate it should be (to evenly distribute samples across the iterators):
```
remainder = size % args.world_size;
if (local_rank < remainder) {
  return (size / world_size) * local_rank + local_rank;
} else {
  return (size / world_size) * local_rank + remainder;
}
```
On the other hand this https://github.com/NVIDIA/DALI/blob/master/dali/python/nvidia/dali/plugin/pytorch.py#L154:
```
  if self._counter >= self._size:
    if self._auto_reset:
      self.reset()
  raise StopIteration
```
You still gets a full batch even if size is not equal.
